tests/Tests/500__jupyterhub/test-jupyterlab-time.robot

```diff
@@ -0,0 +1,87 @@
+*** Settings ***
+Resource            ../../Resources/ODS.robot
+Resource            ../../Resources/Common.robot
+Resource            ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource            ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Library             ../../../../libs/Helpers.py
+Library             DateTime
+Library             OpenShiftCLI
+Library             DebugLibrary
+
+Suite Setup         Load Spawner Page
+Suite Teardown      End Web Test
+
+
+*** Variables ***
+@{LIST_OF_IMAGES}       s2i-minimal-notebook    s2i-generic-data-science-notebook
+...                     pytorch                 tensorflow    minimal-gpu
+
+${LIMIT_TIME} =    40
+
+*** Test Cases ***
+Verify Average Spawn Time Is Less Than 40 Seconds
+    [Documentation]    Verifies that average spawn time for all JupyterHub images is less than 40 seconds
+    [Tags]    ODS-691
+    ...       Tier2
+    ${total_time} =    Get The Total Time For Spawning
+    ${len} =    Get Length    ${LIST_OF_IMAGES}
+    ${total_avg} =    Evaluate    ${total_time} / ${len}
+    Log    total_avg time to spawn ${total_avg}
+    ${result} =    lt    ${total_avg}.0    ${LIMIT_TIME}.0.0
+    Run Keyword Unless    ${result}    Fail
```

Comment:
this could be wrapped into a more self-explanatory kw like `Average Spawning Time Should Be Less Than`